src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -92,7 +92,12 @@ public function handle(GetEditableCombinationsList $query): CombinationListForEd
     {
         $productId = $query->getProductId();
         $product = $this->productRepository->get($productId);
-        $combinations = $this->combinationRepository->getProductCombinations($productId, $query->getLimit(), $query->getOffset());
+
+        $combinations = $this->combinationRepository->getProductCombinations(
+            $productId,
+            $query->getLimit(),
+            $this->countOffset($query->getLimit(), $query->getPage())
```

Comment:
I think this offset conversion should be handled by the Repository in a protected method from `AbstractObjectModelRepository` because it will likely be useful to other repositories as well

Or if you think that the repository should only rely on an `$offset` parameter instead of `$page` (which does make sense) At least this small piece of code could be shared in a Tools class or somewhere else I don't know